app/reporter/reporter.go

```diff
@@ -76,16 +101,41 @@ func (l Reporter) activate() {
 	for {
 		select {
 		case entry := <-l.messages:
-			buffer = append(buffer, entry)
+			// don't save right away, wait for antispam checks
+			time.Sleep(l.saveDelay)
+
+			if !l.messageExists(entry.MessageID) {
+				log.Printf("[DEBUG] message %d has been deleted, skipping", entry.MessageID)
+				continue
+			}
+
+			buffer = append(buffer, entry.Data)
 			if len(buffer) >= 100 { // forced flush every 100 records
 				if err := writeBuff(); err != nil {
 					log.Printf("[WARN] failed to write reporter buffer, %v", err)
 				}
 			}
-		case <-time.After(time.Second * 5): // flush on 5 seconds inactivity
+		case <-ticker.C: // flush on 5 seconds inactivity
 			if err := writeBuff(); err != nil {
 				log.Printf("[WARN] failed to write reporter buffer, %v", err)
 			}
 		}
 	}
 }
+
+//go:generate moq -out mock_reporter.go . httpClient
+
+type httpClient interface {
+	Get(url string) (*http.Response, error)
+}
+
+func (l *Reporter) messageExists(msgID int) bool {
+	resp, err := l.httpCl.Get(fmt.Sprintf("https://t.me/%s/%d?single", l.chatID, msgID))
+	if err != nil {
+		log.Printf("[WARN] failed to check message existence, %v", err)
+		return false
+	}
+	defer resp.Body.Close() // nolint
+
```

Comment:
Not sure how real the issue with HTTP calls like this is one, but with bot API calls, I have seen a need to have some repeater in case of other status codes, all kinds of 5xx.

Reply:
added repeater in here, couldn't use the repeater middleware as it was failing on specified status codes, when I need to fail on unspecified ones